syntax = "proto3";

package fort.myrmidon.pb;

import "google/protobuf/timestamp.proto";


message TrackingDataDirectory {
	string Path       = 1;
	uint64 StartFrame = 2;
	uint64 EndFrame   = 3;

	google.protobuf.Timestamp StartDate = 4;
	google.protobuf.Timestamp EndDate   = 5;
}

message Experiment {
	repeated TrackingDataDirectory DataDirectory = 1;

	string Name    = 2;
	string Author  = 3;
	string Comment = 4;
};


message FramePointer {
	string Path  = 1;
	uint64 Frame = 2;

	google.protobuf.Timestamp PathStartDate = 3;
};

message Estimate {
	string FromFile = 1;

	double XHead = 2;
	double YHead = 3;
	double XTail = 4;
	double YTail = 5;

	double XPosition     = 6;
	double YPosition     = 7;
	double ThetaPosition = 8;
}

message Identification {
	double X     = 1;
	double Y     = 2;
	double Theta = 3;
    int32  ID    = 4;

	repeated Estimate Estimate = 5;

	FramePointer StartFrame = 6;
	FramePointer EndFrame   = 7;
};

message Metadata {
	string JSONData        = 1;
	uint64 StartValidFrame = 2;
}




message AntMetadata {
	int32                   ID       = 1;
	repeated Identification Marker   = 3;
	repeated Metadata       Metadata = 4;
};

message FileHeader {
	int32 MajorVersion = 1;
	int32 MinorVersion = 2;
};


message FileLine {
	Experiment  Experiment = 1;
	AntMetadata AntData    = 2;
};